test/contracts/root-element.spec.tsx

```diff
@@ -0,0 +1,56 @@
+import React = require('react');
+import {findDOMNode} from 'react-dom';
+import {ClientRenderer, expect, RenderingContext} from 'test-drive-react';
+import * as WixReactComponents from '../../src';
+import {isReactComponent} from '../utils/is-react-component';
+
+// const allComponents = Object.keys(WixReactComponents);
+// Not all components are passing the contract test right now.
+// failing: TreeView, NumberInput, Toggle, BirthdayPicker, RadioButton, RadioGroup, Portal
+
+const testedComponents: string[] = ['DatePicker', 'CheckBox', 'Image', 'DropDown'];
+
+describe('Root Element contract', function() {
+    testedComponents.forEach(exportName => describe(exportName, function() {
+        const ComponentClass = (WixReactComponents as any)[exportName];
+        if (isReactComponent(ComponentClass)) {
+            assertRootElementContract(ComponentClass);
+        }
+    }));
+});
+
+export function assertRootElementContract(Component: React.ComponentType<any>): void {
+    const clientRenderer = new ClientRenderer();
+    function render<P>(element: React.ReactElement<P>): RenderingContext<P> & { rootNode: Element } {
+        const output = clientRenderer.render(element);
+        return {rootNode: findDOMNode(output.result as React.ReactInstance), ...output};
+    }
+
+    afterEach(() => {
+        clientRenderer.cleanup();
+    });
+
+    it('performs data-automation-id merge', function() {
```

Comment:
CONTRACT_TEST is the parent level DAID. The component root level DAID is component-dependent, so I can't write a generic test for it. Also this test doesn't have to cover the whole functionality, just test whether it's there in general.